src/main/resources/templates/ocaml/test/run.sh

```diff
@@ -1,27 +1,96 @@
 #!/bin/bash
 
-OUTPUT_FOLDER="test-reports"
-OUTPUT_FILE="$OUTPUT_FOLDER/results.xml"
-ERROR_LOG="build_errors.txt"
-TIMEOUT="5m"  # Timeout for running all tests
-VERBOSE=false # Set to true to print out compile errors. Be aware that compile errors might show test code.
+# copy code from the assignment or solution to the appropriate test folder
+cp_code()
+{
+    cd "$1" || exit
+    rm ./*.ml >/dev/null 2>&1
+    # shellcheck disable=SC2086
+    ln -s $BUILD_ROOT/$1/src/*.ml ./
+    cd ..
+}
 
+OUTPUT_FILE="test-reports/results.xml"
+BUILD_TIMEOUT="1m"
+TEST_TIMEOUT="10m"
+SAFE=false
```

Comment:
Should this really be `false` by default?

Reply:
Safe mode is intended to be used when testing submissions on artemis, while the default is for developing tests locally.
Bamboo will always call the script with the -s argument activating safe mode.
The concern is that if safe was the default and you had to pass some argument to disable it, someone would forget to pass it while developing tests, causing safe mode to delete tests and solution and leading to data loss (safe mode does so to prevent students from cheating by accessing it somehow)

By default (when developing) hidden tests are run, to be able to test them, otherwise you would have to adjust the start/end dates to run all tests and if you dont know the exact time you may set some dummy values and forget to adjust them later, hidden tests could be run during the normal submission time.
The way it works now, you dont have to change things to develop tests and when adding exercises to artemis the hidden tests for the solution will fail, prompting you to adjust the hidden time (before the exerise starts and you know the exact dates).

Optimally artemis would comunicate to the tests if hidden tests should be run but that is not that simple to implement.

Reply:
Alright, that makes sense.